bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/clangd/utils/ClangdUtils.java

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   See git history
+ *******************************************************************************/
+
+package org.eclipse.cdt.lsp.clangd.utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IWorkspace;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Platform;
+
+public class ClangdUtils {
+	public static final String format_file = ".clang-format"; //$NON-NLS-1$
+
+	/**
+	 * Checks if a .clang-format file exists in the workspace root directory. Creates it if not.
+	 * This ensures that the user has a default formatting similar to the CDT default K&R formatter
+	 *
+	 */
+	public void checkWorkspaceClangFormatFile(IWorkspace workspace) {
+		var path = workspace.getRoot().getLocation().append(format_file).toOSString();
+		var formatFile = new File(path);
```

Comment:
This turn from NIO back to IO looks unnecessary, NIO has methods to check existence.

Reply:
I think I've to use the java.io.File class since IFile needs a project. This does not work:
```Java
var formatFile = workspace.getRoot().getFile(new Path(format_file));
```
leads to:
```txt
Caused by: java.lang.IllegalArgumentException: Path must include project and resource name: /.clang-format
```

Reply:
> This does not work:

Yes, sure, it does not. My point was to use `Files#exists` or alike, since you use `Files#copy`
But this is really minor.

Reply:
Sorry sure, I didn't get it :-(

Reply:
I don't know. The version check fails with:
```txt
src/org/eclipse/cdt/lsp/server/ICLanguageServerProvider2.java:19 Invalid @since tag '3.0.0' on org.eclipse.cdt.lsp.server.ICLanguageServerProvider2; expecting two fragments
```
@ruspl-afed or @jonahgraham  anyone an idea which two fragments are meant here???

Reply:
It wants `3.0`, two fragment version while you have 3 fragments

Reply:
Oh, thank you. I haven't seen that :-